### PR TITLE
Feature/leapr

### PR DESCRIPTION
**Pull request by ameliajo**
Added JSON reading for LEAPR. Also removed the requirement that Card20 must exist, since legacy LEAPR compiles even if you give no card20 comments.

### CODE REVIEW

File: src/lipservice/LEAPR/test/LEAPR.test.cpp [hunk 1]
```diff
@@ -137,9 +138,201 @@ SCENARIO( "LEAPR input",
...
+          nlohmann::json refJSON = R"({
+            "nout"  :  20,
+            "title" : "graphite, endf model",
+            "iprint":  0,
+            "nphon" :  101,
+            "mat"   :  31,
+            "za"    :  6012,
+            "isabt" :  1,
+            "ilog"  :  1,
+            "smin"  :  1e-75,
```
Comment (jlconlin) [inline src/lipservice/LEAPR/test/LEAPR.test.cpp:151]: I didn't see `smin` in the NJOY2016 LEAPR documentation. Does that mean the documentation is incorrect?

File: src/lipservice/LEAPR/test/LEAPR.test.cpp [hunk 2]
```diff
@@ -137,9 +138,201 @@ SCENARIO( "LEAPR input",
         REQUIRE( *(card20_1.comment.value) == refCard20_1 );
         REQUIRE( *(card20_2.comment.value) == refCard20_2 );
 
+        AND_THEN( "LEAPR can be turned into JSON" ){
+          nlohmann::json refJSON = R"({
```
Comment (jlconlin) [inline src/lipservice/LEAPR/test/LEAPR.test.cpp:142]: I like this JSON representation. Looks nice and clean.

File: src/lipservice/LEAPR/test/LEAPR.test.cpp [hunk 3]
```diff
@@ -244,6 +437,60 @@ SCENARIO( "LEAPR input",
...
+            "npr"   : 1,
+            "iel"   : 1,
+            "ncold" : 1,
+            "nsk"   : 1,
+            "nss"   : 0,
+            "lat"   : 1,
+            "alpha" : [2.5e-1, 7.5e-1, 1.0],
+            "beta"  : [0.0, 0.2, 0.4, 0.6, 0.8],
+            "temperatures" : [
+              {
```
Comment (jlconlin) [inline src/lipservice/LEAPR/test/LEAPR.test.cpp:462]: Glad you tested one with multiple temperatures.

File: src/lipservice/LEAPR/test/LEAPR.test.cpp [hunk 4]
```diff
@@ -285,7 +532,67 @@ SCENARIO( "LEAPR input",
...
+              "ilog"   : 1,
+              "smin"   : 1e-75,
+              "awr"    : 11.898,
+              "spr"    : 4.7392,
+              "npr"    : 1,
+              "iel"    : 1,
+              "ncold"  : 1,
+              "nsk"    : 1,
+              "nss"    : 1,
+              "b7"     : 1,
```
Comment (jlconlin) [inline src/lipservice/LEAPR/test/LEAPR.test.cpp:554]: Oh good, you remembered to test where there are secondary scatters.

File: src/lipservice/src/LEAPR.hpp [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+}
+
+
+
+
+
+
+
+
+
```
Comment (jlconlin) [inline src/lipservice/src/LEAPR.hpp:224]: Please remove these extra lines.

File: src/lipservice/src/LEAPR.hpp [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+    JSON = nlohmann::json::array();
+  }
+}
+
+inline void to_json( nlohmann::json& JSON, const std::vector<LEAPR::Card20>& card20List ){
+  if ( card20List.size() > 0 ){
+    std::vector<std::string> card20Values(card20List.size());
+    for (size_t i = 0; i < card20List.size(); ++i){
+      card20Values[i] = *(card20List[i].comment.value);
+    }
```
Comment (jlconlin) [inline src/lipservice/src/LEAPR.hpp:192]: range-based for loops are easier than index-based loops.
```suggestion
    for( const auto& card : card20List ){
      card20Values.push_back( *(card.comment.value) );
    }
```

**Review by jlconlin — CHANGES_REQUESTED** [review]
Great work. I really like the tests you have added. That helps know that everything is working properly.

Just a few minor changes to clean things up a bit.

File: src/lipservice/LEAPR/test/LEAPR.test.cpp (hunk 1 above)
Comment (ameliajo) [reply src/lipservice/LEAPR/test/LEAPR.test.cpp:151]: smin is the 5th value on LEAPR Card 4

Comment (jlconlin) [reply src/lipservice/LEAPR/test/LEAPR.test.cpp:151]: Yep, the 2016 manual doesn't have this.

Comment (ameliajo) [reply src/lipservice/LEAPR/test/LEAPR.test.cpp:151]: I might be mistaken, but it looks like it's [here](https://github.com/njoy/NJOY2016-manual/blob/master/njoy16.pdf) on page 668 of the document. Is there a preferred version of the manual we should be using? Sorry about the confusion

Comment (jlconlin) [reply src/lipservice/LEAPR/test/LEAPR.test.cpp:151]: You are correct. I was looking at an older version of the manual.

File: src/lipservice/src/LEAPR.hpp [hunk 7]
```diff
@@ -186,9 +186,9 @@ inline void to_json( nlohmann::json& JSON,
 
 inline void to_json( nlohmann::json& JSON, const std::vector<LEAPR::Card20>& card20List ){
   if ( card20List.size() > 0 ){
-    std::vector<std::string> card20Values(card20List.size());
-    for (size_t i = 0; i < card20List.size(); ++i){
-      card20Values[i] = *(card20List[i].comment.value);
+    std::vector<std::string> card20Values;
```
Comment (jlconlin) [inline src/lipservice/src/LEAPR.hpp:189]: This looks better.

**Review by jlconlin — APPROVED** [review]

**Comment by jlconlin** [comment]
This looks good. We are moving *everything* to the new build system this morning.

**Pull request merged**